share/wake/html/main.js

```diff
@@ -197,11 +197,49 @@ function workspace(node) {
   return res;
 }
 
+function createThemeToggle() {
+  // If the user previously chose a theme, default to that theme.
+  // Otherwise, check the browser/OS preferences for dark vs. light theme.
+  let colorTheme = localStorage?.getItem('wakeColorTheme');
+  colorTheme ||= matchMedia('prefers-color-scheme: dark') ? 'dark' : 'light';
+
+  document.documentElement.className = colorTheme;
+
+  const themeToggle = document.createElement('div');
+  themeToggle.innerHTML = `
+    <label>Light<input type="radio" name="style" value="light">
+    <label>Dark<input type="radio" name="style" value="dark">
+  `;
+
+  const onChange = (e) => {
+    e.preventDefault();
+    document.documentElement.className = e.target.value;
+    if (localStorage) {
```

Comment:
nit: does nil coalescing not work here?

`localStorage?.setItem('wakeColorTheme', e.target.value);`

Reply:
Oh right, yeah, that should work. I wrote the `if` statement when I thought I was going to write something like `localStorage.wakeColorTheme = e.target.value`, and I wasn't sure if that operator worked on the LHS, but then I forgot to change it when I realized that the API was `setItem()` and not a property access.

I just tested this with local storage disabled temporarily on my browser, and it seems to work.

I also discovered a few more things:

- I should probably prefix `localStorage` with `window?.`, since apparently attempting to access an undefined variable will throw a ReferenceError, not return `undefined`. Attempting to access an undefined attribute on an object will return `undefined`, though. This doesn't matter for browsers that support `localStorage` but allow you to disable it, since `localStorage` will be defined and `null` in that case, but I'll do the access off the `window` object just in case.
- I did confirm that `foo?.bar = baz` is illegal.

I'll push up these changes in a sec.

Reply:
Slight edit: I don't need to do `window?.localStorage`, since `window` is always defined in a browser environment, so `window.localStorage?.setItem()` is sufficient.

Side rant: I am so not used to writing non-TypeScript JavaScript and having my editor just tell me when things are undefined/null.